lib/ex_unit/lib/ex_unit/capture_log.ex

```diff
@@ -84,11 +84,14 @@ defmodule ExUnit.CaptureLog do
   end
 
   defp add_capture(pid, opts) do
-    GenEvent.add_mon_handler(Logger, {Console, pid}, {pid, opts})
+    :gen_event.add_sup_handler(Logger, {Console, pid}, {pid, opts})
   end
 
   defp remove_capture(pid) do
-    case GenEvent.remove_handler(Logger, {Console, pid}, nil) do
+    logger_pid = Process.whereis(Logger)
+    result = :gen_event.delete_handler(logger_pid, {Console, pid}, :normal)
+    Process.unlink(logger_pid)
```

Comment:
Eep, we can do this because the logger pid may be linked in test or there could be nested capture logs. I think we will have to spawn a middleman process that adds the handler.